### ISSUE TITLE
latest version: can no longer create a connection

### ISSUE THREAD

**Issue by cc8080** [Borvik/vscode-postgres#97]
I had originally set up a database connection using an earlier version and have been happily working with that for some time.  Today, I tried to help another user get started with this tool and connect to the same database.  We got stuck at the step where it asks what kind of connection to create -- Standard Connection or Secure Connection.  It displays an error and won't proceed.  The errors are shown below.

I decided to remove and reinstall the extension on my machine and now I can't recreate the connection that had been working just fine.

We're both using Windows 10 to talk to a remote database.

Note that I had been seeing notifications about "no pg_hba.conf..." when running queries, but because it would proceed and do the query just fine, I hadn't bothered to figure out the problem.

I am wondering why it's trying to connect to a database called "postgres".  Our database name is not "postgres".  At the point in the process where we get stuck, the tool hasn't yet asked for the database name.

Appreciate any help/workarounds/suggestions.

Error notifications (substituted x's in place of the IP numbers):

Standard connection:

no pg_hba.conf entry for host "xx.xx.xx.xx", user "guest", database "postgres", SSL off

Use Secure Connection

no pg_hba.conf entry for host "xx.xx.xx.xx", user "guest", database "postgres", SSL on

**Comment by Borvik** [comment]
The `postgres` database is the built-in system database that holds the list of databases. It's usually a safe one to connect to if you are either unable or do not know the database you setting up.

The pg_hba.conf error you are seeing is from the postgres server, and is saying the user you are attempting to use to connect to the database doesn't have permission to access the specific database - looks like the postgres database has been locked down for you.

The recent overhaul to the add connection routine is probably at fault.  Prior to this you could create a connection without knowing if it would work, and would have to start over from scratch if expanding the connection in the explorer didn't work.

There might be something I can change to handle this so in case it can't enumerate the databases during setup it prompts for a manually entered one.

**Comment by Borvik** [comment]
This should be addressed in v1.1.10

**Comment by cc8080** [comment]
Thanks so much for the fast response!  I took over this database and am still learning about postgres, and your comments helped me find a workaround.  I noticed that the postgres default database had no access permissions at all, so I granted connect permissions and was able to then create a new connection.  It seems from reading more documentation that this is the "usual" setup. I don't know why this instance was set up differently.  It does seem like adding a way to select which database to connect to would be helpful, particularly for people who can't change the database setup.
Thanks again, and thanks for the excellent extension! It saves me untold hours when creating/debugging queries!
